.github/workflows/build-test.yml

```diff
@@ -0,0 +1,62 @@
+name: build-test
+
+on: [push, pull_request]
+
+concurrency: 
+  # this ensures after each commit the old jobs are cancelled and the new ones
+  # run instead.
+  group: ${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  all-deps:
+    # this job installs all dependencies and runs most tests
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python: ["3.8", "3.9", "3.10"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python }}
+      - run: pip install .
+      - run: pip install -r requirements-dev.txt
+      - run: pytest --ignore=test/install --cov=fairlearn --cov-report=xml -m "not notebooks" test/
```

Comment:
Quite a few of these steps appear to be shared.... can that be abstracted?

Reply:
Except the codecov upload, none of them are exactly the same, and having them in the same file would mean accepting input args and conditioning on those and at the end it'd be more code than what it is at this stage. I don't want to overengineer things at this stage, happy to abstract things away once it starts to actually save lines of code.

Reply:
Having worked with the previous system a bit I will say I'd take a bit of duplication if it gives us readability (as in, being able to read the entire workflow end-to-end in one file).